internal/controller/storageclient_controller.go

```diff
@@ -235,6 +235,16 @@ func (r *StorageClientReconciler) reconcilePhases() (ctrl.Result, error) {
 					return reconcile.Result{}, fmt.Errorf("failed to reconcile cephConnection: %v", err)
 				}
 			}
+		case "Subscription":
+			var subscriptionSpec *opv1a1.SubscriptionSpec
+			if err := json.Unmarshal(eResource.Data, &subscriptionSpec); err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed to unmarshall subscriptionSpec error: %v", err)
+			}
+
+			utils.AddAnnotation(r.storageClient, utils.DesiredSubscriptionChannelAnnotationKey, subscriptionSpec.Channel)
```

Comment:
this could be moved into a conditional, iirc `utils.AddAnnotation` return `true` if in-memory object is updated.

Reply:
Done